pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -793,12 +795,48 @@ public void openLedgerFailed(ManagedLedgerException exception, Object ctx) {
                     managedLedgerConfig.setOffloadAutoTriggerSizeThresholdBytes(p.offload_threshold);
                 });
 
+
+            try {
+                managedLedgerConfig.setLedgerOffloader(createManagedLedgerOffloader(serviceConfig, offloadPolicies));
```

Comment:
@sijie If I'm correct, it's done here.

Reply:
But then I don't know which ledgerFactory should be used, not sure about what their job :/ can you explain them?

Reply:
@jiazhai maybe? :D

Reply:
poke

Reply:
@sijie @jiazhai poke